### PR TITLE
bug #450: fixed bug on teacher role OTHER

### PR DESCRIPTION
**Pull request by sifes**
- removed all usages of Teacher Roles enum in DTOs, Responses and other API files
- replaced Teacher Roles by Discipline Types
- removed Teacher Roles enum from database

closes [#450](https://github.com/fictadvisor/fictadvisor/issues/450)

### CODE REVIEW

File: fictadvisor-web/.env.development [hunk 1]
```diff
@@ -1,5 +1,5 @@
-NEXT_PUBLIC_API_BASE_URL=https://apidev.ficeadvisor.com/v2
-API_BASE_URL=https://apidev.ficeadvisor.com/v2
+NEXT_PUBLIC_API_BASE_URL=http://localhost:4455/v2
+API_BASE_URL=http://localhost:4455/v2
```
Comment (sifes) [inline fictadvisor-web/.env.development:2]: do not forget to change back this

File: fictadvisor-web/src/app/admin/disciplines/search/constants/index.ts [hunk 2]
```diff
@@ -19,7 +19,8 @@ export const sortOptions: DropDownOption[] = [
 
 export const disciplineTeacherInitialValues = {
   id: '',
-  roles: [],
+
```
Comment (sifes) [inline fictadvisor-web/src/app/admin/disciplines/search/constants/index.ts:22]: remove space

File: fictadvisor-web/src/app/admin/groups/edit/[groupId]/constants/index.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import { TeacherRole } from '@fictadvisor/utils/enums';
+import { DisciplineTypeEnum } from '@fictadvisor/utils';
 
 export const rolesOptions = [
```
Comment (sifes) [inline fictadvisor-web/src/app/admin/groups/edit/[groupId]/constants/index.ts:3]: no more rolesOptions

File: fictadvisor-web/src/app/admin/groups/edit/[groupId]/constants/index.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
-  { value: TeacherRole.LABORANT, label: 'Лаборант' },
-  { value: TeacherRole.LECTURER, label: 'Викладач' },
-  { value: TeacherRole.OTHER, label: 'Інше' },
-  { value: TeacherRole.PRACTICIAN, label: 'Практик' },
+  { value: DisciplineTypeEnum.EXAM, label: 'Екзаменатор' },
+  { value: DisciplineTypeEnum.LABORATORY, label: 'Лаборант' },
+  { value: DisciplineTypeEnum.LECTURE, label: 'Викладач' },
+  { value: DisciplineTypeEnum.PRACTICE, label: 'Практик' },
+  { value: DisciplineTypeEnum.CONSULTATION, label: 'Консультант' },
+  { value: DisciplineTypeEnum.WORKOUT, label: 'Інше' },
```
Comment (sifes) [inline fictadvisor-web/src/app/admin/groups/edit/[groupId]/constants/index.ts:9]: there is no 'Інше' now

File: fictadvisor-web/src/app/admin/teachers/edit/[teacherId]/constants/index.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
-import { TeacherRole } from '@fictadvisor/utils/enums';
+import { DisciplineTypeEnum } from '@fictadvisor/utils';
 
 export const rolesOptions = [
```
Comment (sifes) [inline fictadvisor-web/src/app/admin/teachers/edit/[teacherId]/constants/index.ts:3]: no more rolesOptions

File: fictadvisor-web/src/app/admin/teachers/search/components/teachers-admin-search/TeachersAdminSearch.tsx [hunk 6]
```diff
@@ -45,15 +44,11 @@ const TeachersAdminSearch: FC<TeachersAdminSearchProps> = ({ onSubmit }) => {
...
   const handleFormSubmit = () => {
     const newCathedrasId = cathedrasId.map(cathedra => cathedra.id) as string[];
-    // Failed Logic!!!
-    const newRoles = roles.map(role => role.id) as TeacherRole[];
-    // as TeacherRole[] temporary kostil!!!
     setValues({
       ...values,
       search,
       order,
       cathedrasId: newCathedrasId,
```
Comment (sifes) [inline fictadvisor-web/src/app/admin/teachers/search/components/teachers-admin-search/TeachersAdminSearch.tsx:51]: where is the related logic for disciplineType

File: fictadvisor-web/src/app/admin/teachers/search/constants/index.ts [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
-  roles: [],
 };
 
-export const teacherRoles: TeacherRole[] = [
-  TeacherRole.LECTURER,
-  TeacherRole.LABORANT,
-  TeacherRole.PRACTICIAN,
-  TeacherRole.EXAMINER,
-  TeacherRole.OTHER,
+export const teacherRoles: DisciplineTypeEnum[] = [
```
Comment (sifes) [inline fictadvisor-web/src/app/admin/teachers/search/constants/index.ts:11]: there is no teacherRoles

File: fictadvisor-api/src/v2/mappers/TeacherMapper.ts [hunk 8]
```diff
@@ -1,17 +1,16 @@
 import { Injectable } from '@nestjs/common';
 import { Sort, SortDTO } from '@fictadvisor/utils/requests';
 import {
-  TeacherRole,
   OrderQAParam,
   SortQATParam,
   AcademicStatus,
   ScientificDegree,
-  Position,
+  Position, DisciplineTypeEnum,
```
Comment (dmykyr) [inline fictadvisor-api/src/v2/mappers/TeacherMapper.ts:8]: move to next row

**Review by dmykyr — CHANGES_REQUESTED** [review]

**Review by dmykyr — APPROVED** [review]

**Review by foxsis23 — APPROVED** [review]

**Pull request merged**